aria_data_loaders/aria_data_utils/aria_streamer.py

```diff
@@ -0,0 +1,842 @@
+            self.device_calib.get_transform_device_sensor(stream_name).to_matrix()
+        )
+        assert device_T_camera is not None
+
+        # Setup camera calibration parameters by over-writing self._src_calib_param & self._dst_calib_param
+        if should_rectify:
+            self._src_calib_params = self.device_calib.get_camera_calib(stream_name)
+            self._dst_calib_params = calibration.get_linear_camera_calibration(
+                512, 512, 280, stream_name
+            )
+
+        # Setup April tag detection by over-writing self._qr_pose_estimator
+        if detect_qr:
+            self._init_april_tag_detector()
+
+        if should_display:
+            self._create_display_window(stream_name)
+
+        num_frames = self.provider.get_num_data(stream_id)
+        # CUSTOM RANGE FOR VIDEO - TODO: REMOVE LATER
+        # custom_range = range(1400, 1550)
+        custom_range = range(0, num_frames)
+        for frame_idx in custom_range:
+            frame_data = self.provider.get_image_data_by_index(stream_id, frame_idx)
+            img = frame_data[0].to_numpy_array()
+            img_metadata = frame_data[1]
+
+            camera_T_marker = None
+
+            if should_rectify:
+                img = self._rectify_image(image=img)
+
+            if detect_qr:
+                (
+                    img,
+                    camera_T_marker,
+                ) = self._qr_pose_estimator.detect_markers_and_estimate_pose(  # type: ignore
+                    image=img, should_render=True, magnum=False
+                )
+
+            img = self._rotate_img(img=img)
+
+            if camera_T_marker is not None:
+                cpf_T_marker = self.cpf_T_device * device_T_camera * camera_T_marker
+                img = decorate_img_with_text(
+                    img=img,
+                    frame="cpf",
+                    position=cpf_T_marker.translation(),
+                )
+                print(
+                    f"Time stamp with Detections- {img_metadata.capture_timestamp_ns}"
+                )
+
+                img_list.append(img)
+                img_metadata_list.append(img_metadata)
+                cpf_T_marker_list.append(cpf_T_marker)
+
+            if should_display:
+                self._display(img=img, stream_name=stream_name)
+
+            if should_rec_timestamp_from_user:
+                val = input("o to skip, c to record")
+                if val == "c":
+                    self.vrs_idx_of_interest_list.append(frame_idx)
+
+        return img_list, img_metadata_list, cpf_T_marker_list
+
+    def initialize_trajectory(self):
+        # frame(ariaWorld) is same as frame(device) at the start
+        cpf_T_ariaWorld = self.cpf_T_device
+
+        for i in range(len(self.mps_trajectory)):
+            self.trajectory_s[i] = self.mps_trajectory[
+                i
+            ].tracking_timestamp.total_seconds()
+            ariaWorld_T_device = sp.SE3(
```

Comment:
Should L320, L325-327 be a function of its own?

Reply:
Made some clean-ups, can you recommend changes here again if needed?